search_log.go

```diff
@@ -136,34 +176,37 @@ func readLine(reader *bufio.Reader) (string, error) {
 	return string(line), nil
 }
 
-func readLastLine(file *os.File) string {
+// Read a line from the end of a file
+// startCursor initial value should be the filesize
+// TODO: read byte by byte is low efficiency, we can improve it, for example, read 1024 bytes one time
+func readLineReverse(file *os.File, startCursor int64) string {
 	var line []byte
-	var cursor int64
-	stat, _ := file.Stat()
-	filesize := stat.Size()
+	var cursor = startCursor
 	for {
-		cursor -= 1
-		file.Seek(cursor, io.SeekEnd)
+		// stop if we are at the begining
+		// check it in the start to avoid read beyond the size
+		if cursor <= 0 {
+			break
+		}
 
+		cursor--
+		file.Seek(cursor, io.SeekStart)
```

Comment:
Replace `file.Seek(cursor, io.SeekEnd)` by `file.Seek(cursor, io.SeekStart)`, the former way may read incorrect content from the end of a file when some others append new logs to this file.